pkg/threescale/backend/backend.go

```diff
@@ -421,19 +419,29 @@ func (b *Backend) GetPeer() string {
 	return b.client.GetPeer()
 }
 
+// calculateDeltas returns a set of metrics that we can report to 3scale using the lowest period known in the counters
+// for metrics with rate limits attached
 // calculateDeltas returns a set of metrics that we can report to 3scale using the lowest period known in the counters
 // for metrics with rate limits attached
 func (a *Application) calculateDeltas() api.Metrics {
 	deltas := make(api.Metrics)
 
-	for metric := range a.LimitCounter {
-		lowestPeriod := a.LimitCounter.getLowestKnownPeriodForMetric(metric)
-		if lowestPeriod != nil {
-			if lastKnownState, ok := a.RemoteState[metric][*lowestPeriod]; ok {
-				toReport := a.LimitCounter[metric][*lowestPeriod].current - lastKnownState.current
-				deltas.Add(metric, toReport)
-			}
+	for metric, counters := range a.LocalState {
+		// we want to handle the lowest granularity only here
```

Comment:
In this case this is right, because the deltas we want to report to 3scale only need the hits that have occurred since the last time we flushed, and 3scale will apply the deltas to all granularities.